src/LIBSVM.jl

```diff
@@ -373,7 +379,9 @@ function svmpredict(model::SVM{T}, X::AbstractMatrix{U}; nt::Integer = 0) where
     set_num_threads(nt)
 
     if model.kernel != Kernel.Precomputed && size(X, 1) != model.nfeatures
-        throw(DimensionMismatch("Model has $(model.nfeatures) but $(size(X, 1)) provided"))
+        throw(DimensionMismatch("Model has $(model.nfeatures) features but $(size(X, 1)) provided"))
+    elseif model.kernel == Kernel.Precomputed && size(X, 1) != model.nfeatures
+            throw(DimensionMismatch("Gram matrix should either have $(model.nfeatures) or $(sum(model.SVs.nSV)) features but $(size(X, 1)) provided"))
```

Comment:
Don't forget to update this error message too (since you decided not to allow the "number-of-support-vectors-optimization")

Reply:
yep, just realized that as well - my bad.